src/test/java/com/redhat/devtools/lsp4ij/features/foldingRange/TypeScriptCodeBlockProviderTest.java

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.foldingRange;
+
+import com.redhat.devtools.lsp4ij.fixtures.LSPCodeBlockProviderFixtureTestCase;
+
+/**
+ * Selection range tests by emulating LSP 'textDocument/foldingRange' responses from the typescript-language-server.
+ */
+public class TypeScriptCodeBlockProviderTest extends LSPCodeBlockProviderFixtureTestCase {
+
+    private static final String DEMO_TS_FILE_NAME = "demo.ts";
+    // language=typescript
+    private static final String DEMO_TS_FILE_BODY = """
+            export class Demo {
+                demo() {
+                    console.log('demo');
+                }
+            }
+            """;
+
+    public TypeScriptCodeBlockProviderTest() {
+        super("*.ts");
+    }
+
+    public void testCodeBlocks() {
+        assertCodeBlocks(
+                DEMO_TS_FILE_NAME,
+                DEMO_TS_FILE_BODY,
+                // language=json
+                """
+                        [
+                          {
+                            "startLine": 0,
+                            "endLine": 3
+                          },
+                          {
+                            "startLine": 1,
+                            "endLine": 2
+                          }
+                        ]
+                        """,
+                new int[][]{
+                        // From "console" to the begin and end of the "demo()" function's braced block
```

Comment:
Instead of expect some offset, I think will more readable to expect a string which have the same content of the editor but with some annotations like <fold1> ends with </fold1>

It follows the same idea with the <caret> that yiu can use in a test (see for instance completion test, where I use <caret> instead the offset number

Reply:
Here a sample with folding ranges:

```java
public void testFoldingRanges() {
        assertFoldingRanges(
                "demo.ts",
                """
                export class Demo {
                    demo() {
                        console.log('demo');
                    }
                }
                """,
                // language=json
                """
                        [
                          {
                            "startLine": 0,
                            "endLine": 3
                          },
                          {
                            "startLine": 1,
                            "endLine": 2
                          }
                        ]
                        """,
                """
                <fold1>export class Demo {
                    demo() {
                        console.log('demo');
                    }
                }</fold1>
                """
        );
    }
}
```

Reply:
I've changed the tests to provide a balance between readability and maintainability. Tokens are now used, but directly in the test file source, and they're both extracted for offset information and stripped from the file body for the actual `PsiFile` creation. That way you're only maintaining one version of the file body with tokens representing key locations.

Let me know if that doesn't address your concerns.